packages/did/src/Did.utils.ts

```diff
@@ -380,3 +380,21 @@ export function assembleKeyUri(
   }
   return `${did}#${keyId}`
 }
+
+export function isUri(s: string): boolean {
+  try {
+    const url = new URL(s) // this actually accepts any URI but throws if it can't be parsed
```

Comment:
hmm, what are the pros of using `new URL` here instead of a regex?

Reply:
that you don't have to define a regex that is so long and convoluted that nobody understands it. Try defining the regex for a valid URI

Reply:
(your regex would not just have to limit the character set but reproduce URI syntax)